x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/ApiKeyService.java

```diff
@@ -255,6 +258,22 @@ public void createApiKey(Authentication authentication, CreateApiKeyRequest requ
         }
     }
 
+    public boolean isInEnrollmentMode () {
```

Comment:
As said before I think we should revert the changes in ApiKeyService and ApiKeyGenerator. `isInEnrollmentMode` should be checked just once, in the transport action before we even reach ApiKeyService or ApiKeyGenerator